presto-spark-base/src/main/java/com/facebook/presto/spark/planner/PrestoSparkRddFactory.java

```diff
@@ -476,4 +515,42 @@ private static void checkInputs(
     {
         return scala.reflect.ClassTag$.MODULE$.apply(clazz);
     }
+
+    private static class SparkPartition
+            implements Comparable<SparkPartition>
+    {
+        private final int maxSplitSlot;
+        private final int partitionId;
+        private int usedSplitSlot;
+        private long splitsSizeInBytes;
+
+        public SparkPartition(int partitionId, int minSplitsPerSparkPartition)
+        {
+            this.partitionId = partitionId;
+            this.maxSplitSlot = minSplitsPerSparkPartition;
+        }
+
+        @Override
+        public int compareTo(SparkPartition o)
+        {
+            if (usedSplitSlot != o.usedSplitSlot) {
+                return usedSplitSlot - o.usedSplitSlot;
+            }
+            return splitsSizeInBytes == o.splitsSizeInBytes ?
```

Comment:
I would simply sort it by size. We should always add a next split to the partition that has the least data assigned (not the least number of splits).

Reply:
Make sense,  `spark.min-splits-per-partition` is only used to determine total number of spark task:  avoid spawn new task that has few splits.  In most cases,  spark task count should be determined by data size per task.

Reply:
simply `this. splitsSizeInBytes  - that.splitsSizeInBytes`

Reply:
This avoids overflow when converting Long to Int.